Source/WebCore/platform/graphics/cocoa/WebCoreCALayerExtras.mm

```diff
@@ -119,6 +123,20 @@ - (BOOL)_web_maskMayIntersectRect:(CGRect)rect
     return CGRectIntersectsRect(self.mask.bounds, rectInMask);
 }
 
+- (void)_web_clearContents
+{
+    self.contents = nil;
+    self.contentsOpaque = NO;
+
+#if ENABLE(CG_DISPLAY_LIST_BACKED_IMAGE_BUFFER)
+    if ([self valueForKeyPath:WKCGDisplayListContentsKey]) {
+        [self setValue:nil forKeyPath:WKCGDisplayListContentsKey];
+        [self setValue:nil forKeyPath:WKCGDisplayListPortsKey];
+        [self setNeedsDisplay];
```

Comment:
This is a bit odd. Often you clear contents to remove backing store, but calling `-setNeedsDisplay` will trigger a `-display` which will reallocate backing store.

Reply:
good... point

Reply:
Can't remove the setNeedsDisplay yet, but added a workaround that will prevent backing store from being allocated for now.